pkgs/java_http/lib/src/java_client.dart

```diff
@@ -78,12 +84,27 @@ class JavaClient extends BaseClient {
         reasonPhrase: reasonPhrase);
   }
 
-  int _statusCode(BaseRequest request, HttpURLConnection httpUrlConnection) {
+  void _setRequestBody(
+    HttpURLConnection httpUrlConnection,
+    Uint8List requestBody,
+  ) {
+    if (requestBody.isEmpty) return;
+
+    httpUrlConnection.setDoOutput(true);
+    final outputStream = httpUrlConnection.getOutputStream();
+    requestBody.forEach(outputStream.write);
+
+    outputStream
```

Comment:
[optional] Is there any value in minimizing the uses of `callMethodWithArgs` and avoiding interop calls in a loop?

(untested suggestion)

```suggestion
    httpUrlConnection.setDoOutput(true);
    final outputStream = httpUrlConnection.getOutputStream();
    final bodyArray = JArray(jbyte.type, requestBody.length)
      ..setRange(0, requestBody.length, requestBody);
    outputStream
      ..write1(bodyArray)
```

Reply:
Sounds like a good idea to me!

I've pushed a commit with the suggestions and the tests were all passing locally.

Thank you :smile:

Reply:
I've been learning about extension methods recently, could we add an extension method to `Uint8List`?

```dart
extension on Uint8List {
  JArray<jbyte> toJArray() =>
      JArray(jbyte.type, length)..setRange(0, length, this);
}
```

Then, the code would go from:
```dart
    final bodyArray = JArray(jbyte.type, requestBody.length)
      ..setRange(0, requestBody.length, requestBody);

    httpUrlConnection.getOutputStream()
      ..write1(bodyArray)
      ..flush()
      ..close();
```

to:

```dart
    httpUrlConnection.getOutputStream()
      ..write1(requestBody.toJArray())
      ..flush()
      ..close();
```

My intuition is that adding an extension method to `Uint8List` is unnecessary at the moment. But if it becomes a common occurence to convert `Uint8List` to `JArray<jbyte>` in `java_http` then it would be a nice idea to add an extension method.

Reply:
> could we add an extension method to `Uint8List`?

This extension method does look like a good idea.

Reply:
Awesome! Let's go for the extension method `toJArray()` on `Uint8List`.